### PR TITLE
CI/CD Improvements/fixes

### PR DESCRIPTION
**Pull request by stevenewald**

### CODE REVIEW

File: .github/workflows/linter-cicd.yml [hunk 1]
```diff
@@ -75,6 +75,9 @@ jobs:
       uses: actions/setup-python@v4
       with: { python-version: "3.11" }
 
+    - name: Install Python Developer Headers
+      run: sudo apt install python3.11-dev
```
Comment (egelja) [inline .github/workflows/linter-cicd.yml:79]: Uhh don't do this use the install Python action they have an option for headers iirc

File: linter/Dockerfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ FROM python:3.11-slim as build
 RUN pip install conan numpy pandas polars scipy scikit-learn \
     # c++ stuff
     && apt update \
-    && apt install -y --no-install-recommends build-essential libssl-dev cmake git
+    && apt install -y --no-install-recommends build-essential libssl-dev cmake git python3.11-dev
```
Comment (egelja) [inline linter/Dockerfile:7]: Headers are already installed this is unneeded.

**Review by egelja — CHANGES_REQUESTED** [review]

File: linter/Dockerfile (hunk 2 above)
Comment (stevenewald) [reply linter/Dockerfile:7]: ah

File: .github/workflows/linter-cicd.yml (hunk 1 above)
Comment (stevenewald) [reply .github/workflows/linter-cicd.yml:79]: ok will check

**Pull request merged**